components/shared/BalanceInput.vue

```diff
@@ -88,16 +81,18 @@ export default class BalanceInput extends mixins(ChainMixin) {
   }
 
   @Debounce(200)
-  @Emit('input')
   public handleInput(value: number) {
     this.internalValue = value
+    this.$emit('input', String(this.internalValue * this.selectedUnit))
     return String(value)
   }
 
   handleUnitChange(unit) {
-    this.selectedUnit = unit
+    const valueInBSXUnit = this.internalValue * this.selectedUnit
```

Comment:
dafuq

Reply:
unit exchange.

e.g., input as 10000 BSX <-> 10 Kilo while unit change. Firstly convert all units into BSX unit first and then convert to the selected unit.

Reply:
I am not sure if BSX sounds like share imo

<img width="269" alt="Screenshot 2022-07-27 at 20 13 17" src="https://user-images.githubusercontent.com/22471030/181343101-ac166c78-db7b-4d96-b045-3948e4dafb21.png">

Reply:
changed to valueInBaseUnit.

Reply:
thanks ^-^